localstack/aws/spec.py

```diff
@@ -51,13 +52,14 @@ def signing_name_index(self) -> Dict[str, List[ServiceName]]:
         return dict(result)
 
     @cached_property
-    def operations_index(self) -> Dict[str, List[ServiceName]]:
-        result = defaultdict(list)
+    def protocol_and_operations_index(self) -> Dict[ServiceProtocol, Dict[str, List[ServiceName]]]:
+        result = defaultdict(lambda: defaultdict(list))
```

Comment:
se comment above. i think we should leave the underlying abstraction (op -> service mapping) unchanged here and leave the protocol filtering up to the caller.